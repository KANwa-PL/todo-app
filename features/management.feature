Feature: TODO Management

  Scenario: I cannot add an empty TODO

    Given I am on the main page
    Then I should see "No Todos" message displayed
    When I add a new TODO with the title ""
    Then I should see a notification "Please enter a title"
    And I close the form

  Scenario: I can add a new TODO

    When I add a new TODO with the title "Buy Milk"
    Then I should see a notification "Task added successfully"
    And I should see the "incomplete" TODO

  Scenario: I can edit existing TODO

    When I update a TODO with a new title "Buy milk and some cereals"
    And I update a TODO with a new status "completed"
    Then I should see a notification "Task Updated successfully"
    And I should see the "completed" TODO

  Scenario: I can remove existing TODO

    When I remove a TODO
    Then I should see a notification "Todo Deleted Successfully"
    And I should see "No Todos" message displayed
    