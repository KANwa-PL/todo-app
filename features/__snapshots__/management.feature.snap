// Snapshot v1

exports[`I can add a new TODO > I should see the "incomplete" TODO 1`] = `
"<div class="todoItem_svgBox__z1vm6" style="background: rgb(222, 223, 225);">
  <svg viewBox="0 0 53 38" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path fill="none" stroke-miterlimit="10" stroke-width="6" d="M1.5 22L16 36.5L51.5 1" stroke-linejoin="round" stroke-linecap="round" opacity="0" pathLength="1" stroke-dashoffset="0px" stroke-dasharray="0px 1px"></path>
  </svg>
</div>"
`;

exports[`I can edit existing TODO > I should see the "completed" TODO 1`] = `
"<div class="todoItem_svgBox__z1vm6" style="background: rgb(100, 111, 240);">
  <svg viewBox="0 0 53 38" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path fill="none" stroke-miterlimit="10" stroke-width="6" d="M1.5 22L16 36.5L51.5 1" stroke-linejoin="round" stroke-linecap="round" opacity="1" pathLength="1" stroke-dashoffset="0px" stroke-dasharray="1px 1px"></path>
  </svg>
</div>"
`;
